### PR TITLE
Replace installs reference in RSpace base class from SyncVar to Ref

### LINKED ISSUE #3689: Replace use of SyncVar with Ref (support scala 2.13)
## Overview

In Scala 2.13 SyncVar is removed so we need to switch to alternative. Another reason is to use functional abstraction which is `Ref[F, A]` from cats library.

### Solution

Replace objects defined as `SyncVar` with `Ref` [concurrent mutable reference](https://typelevel.org/cats-effect/docs/2.x/concurrency/ref).
Each item should be in a separate PR.

- [x] Replace objects to store report events in Reporting RSpace.
  #3735
https://github.com/rchain/rchain/blob/2a52bd8623ee4223768002327318fe6c485981fb/rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala#L114-L115
- [x] Replace eventLog reference in RSpace base class.
  #3736
  https://github.com/rchain/rchain/blob/8b9e31582835ccc2ea7260260895fe18ad789e68/rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala#L57
- [x] Replace produceCounters reference in RSpace base class. This one is little more involved. `COMM` constructor should return wrapped COMM object to calculate timesRepeated with F result.
  https://github.com/rchain/rchain/blob/8b9e31582835ccc2ea7260260895fe18ad789e68/rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala#L58
- [x] Replace installs reference in RSpace base class.
  https://github.com/rchain/rchain/blob/8b9e31582835ccc2ea7260260895fe18ad789e68/rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala#L87-L91
- [x] replace SyncVar with LinkedBlockingQueue as lock in Transport Layer test.
  https://github.com/rchain/rchain/blob/aad68bda94f34a8f5858b4518b5b2925d4d1ac9c/comm/src/test/scala/coop/rchain/p2p/EffectsTestInstances.scala#L65-L67

### PR DESCRIPTION
**Pull request by SivkovAV**
## Overview
In Scala 2.13 SyncVar is removed so we need to switch to alternative. Another reason is to use functional abstraction which is Ref[F, A] from cats library.

### Solution
Replace objects defined as SyncVar with Ref [concurrent mutable reference](https://typelevel.org/cats-effect/docs/2.x/concurrency/ref).
Each item should be in a separate PR.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

resolved https://github.com/rchain/rchain/issues/3689 (task 4)

### CODE REVIEW

**Comment by tgrospic** [comment]
Please rebase your PR on the latest dev branch. When CI executed tests it will merge your PR to the latest dev first and it seems the problem is in compile phase.

**Comment by stanislavlyalin** [comment]
There is no need to repeat the description of the task in the description of the pull request.
If the pull request is still in progress, it is better to mark it as a draft

**Review by tgrospic — APPROVED** [review]
LGTM

**Review by nzpr — APPROVED** [review]
LGTM

File: rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala [hunk 1]
```diff
@@ -285,11 +280,7 @@ abstract class RSpaceOps[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C,
         result <- options match {
                    case None =>
                      for {
-                       _ <- syncF.delay {
```
Comment (stanislavlyalin) [inline rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala:288]: Do we really need `syncF` and `spanF` in the codebase?

**Review by stanislavlyalin — APPROVED** [review]

File: rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala (hunk 1 above)
Comment (tgrospic) [reply rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala:288]: Yes, they are defined on the trait this abstract class extends.
https://github.com/rchain/rchain/blob/76f14c516bac3d6be2f91318c9b202b4239465bb/rspace/src/main/scala/coop/rchain/rspace/SpaceMatcher.scala#L17-L22

**Comment by tgrospic** [comment]
bors merge

File: rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala (hunk 1 above)
Comment (stanislavlyalin) [reply rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala:288]: I mean, do we even need it in the trait?

**Pull request merged**